fx quant: add quantized Softmax workflow integration

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#75106 fx quant: add quantized Softmax workflow integration**

Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

vkuzo committed:
eager quant: add quantized Softmax workflow integration

Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the eager mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`. A future PR will
add FX graph mode workflow integration.

Test plan:

```
python test/test_quantization.py TestQuantizeEagerPTQStatic.test_softmax
```

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

jerryzh168 commented on test/quantization/fx/test_quantize_fx.py:
```diff
@@ -6002,6 +6003,8 @@ def forward(self, x):
                 # F.tanh is deprecated
                 x = torch.tanh(x)
                 x = x.tanh()
+                # TODO(future PR): handle F.softmax
```
we can wait a bit and support in the new api: https://github.com/pytorch/pytorch/pull/75135

jerryzh168 replied:
yeah I'm talking about new additions, not this one, please feel free to land this for urgent use cases

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

vkuzo committed:
Update on "fx quant: add quantized Softmax workflow integration"


Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).
